### ISSUE TITLE
Multisig Actor proposal transactions hang for high param counts

### ISSUE THREAD

**Issue by AmeanAsad** [filecoin-project/lotus#10995]
### Context
We are using a multisig actor to propose an invocation of a function on a smart contract. That function takes in an array of addresses as a param. 

### Issue we are facing:
We have noticed that when trying to make proposals where the size of the input array (or param count in general) is large (> 50), the proposal transactions always hangs in a "pending" state. However when using low param count, the proposals always go through without the transaction "pending". 

Will also note that when invoking the contract directly with the same large param counts, the transactions go through without hanging at all. We have tested this numerous times with identical contract invocations.

###  What we have tried:
- We upped our gas multiplier by x10 and still no avail.
- We tried deploying new multisigs to see if this a one off situation. 
- We monitored the network for gas spikes and saw none that justify transactions always going to "pending". 

### Ideal Behaviour:

For multisig proposals to work consistently with large param counts.

**Comment by longfeiWan9** [comment]
@Stebalien @fridrik01 @jennijuju   can you guys please take a look at this issue since it is blocking the payout process that Saturn is working on right now ?

**Comment by Stebalien** [comment]
> This time, even though the messages were processed on chain, their method was shown as CancelExported and the proposal was not visible on the Multisig contract. You can check this [here](https://filfox.info/en/address/f2qkvvjh6ghvlfq6zh4pspsx24fhmnsmf36eumhtq)

1. This would mean you called the `CancelExported` method somewhere and likely didn't call approve.
2. It looks like there's a bug in the go-state-types library as the `Exported` multisig methods were removed from the multisig actor (before they were ever available on the network). They should also have been removed from the go-state-types.

**Comment by Stebalien** [comment]
Are you ever calling approve? I'm only seeing proposal messages and this is a 2/3 multisig.

**Comment by AmeanAsad** [comment]
@Stebalien Our ideal deployment process for our payouts is that we create all the proposals on the multisig then we start issuing approvals. We didn't call approve because we weren't sure that all the proposals would go through after we started having issues and this was on mainnet.

Also where are you seeing a 2/3 multisig? I see three signatories and the approval threshold is 3.

**Comment by AmeanAsad** [comment]
We did and 10x the priority fee but it didn't seem to speed up the proposals. My other question / concern is why is does it take so long for a proposal message that invokes a smart contract on a multisig to be confirmed when the exact same message executes within ~1min when invoking the smart contract directly?

**Comment by Stebalien** [comment]
Just to be very clear, when you say "pending" you're talking about the _filecoin_ messages, not the pending multisig transactions?

To answer your question, I'm not sure. Proposals will be _slightly_ larger but not _much_ larger. The only thing I can think of is that maybe some storage providers are censoring multisig messages, but I haven't heard of anyone doing that.

It's more likely that there's a difference in how messages are getting sent. E.g., some messages may have gotten blocked on another message that had too low of a base-fee, etc. I assume when you invoked the contract directly you were doing this manually with, e.g., metamask (or at least a different path).

**Comment by AmeanAsad** [comment]
> Just to be very clear, when you say "pending" you're talking about the filecoin messages, not the pending multisig transactions?

Yes that is correct. Speaking about a Filecoin message that has not been confirmed yet and NOT a multisig transaction that is waiting to approved by other signatories. 

> I assume when you invoked the contract directly you were doing this manually with, e.g., metamask (or at least a different path).

That is correct, I think I should then test invoking the contract with the same method used to invoke the multisig and observe if those transactions are also "pending" for a while. This should give us a conclusive answer on whether this is related to the method used to send the messages.

**Comment by Stebalien** [comment]
> That is correct, I think I should then test invoking the contract with the same method used to invoke the multisig and observe if those transactions are also "pending" for a while. This should give us a conclusive answer on whether this is related to the method used to send the messages.

That sounds like a good next step. Also, feel free to ping me on slack while this is happening and I can try to walk through it and debug it with you. After the fact, it's a bit difficult to tell exactly what happened.
